core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponentSpec.scala

```diff
@@ -269,6 +269,46 @@ class AttributeComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     }
   }
 
+  it should "unmarshal ints and doubles with appropriate precision" in withEmptyTestDatabase {
+    val attributeRecs = Seq(
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "int", None, Some(2), None, None, None, None, None, false, None)), None),
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "negInt", None, Some(-2), None, None, None, None, None, false, None)), None),
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "doubleRounded", None, Some(2.0), None, None, None, None, None, false, None)), None),
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "negDoubleRounded", None, Some(-2.0), None, None, None, None, None, false, None)), None),
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "double", None, Some(2.34), None, None, None, None, None, false, None)), None),
+      ((1, WorkspaceAttributeRecord(dummyId2, workspaceId, AttributeName.defaultNamespace, "negDouble", None, Some(-2.34), None, None, None, None, None, false, None)), None)
+    )
+
+    val unmarshalled = workspaceAttributeQuery.unmarshalAttributes(attributeRecs)
+
+    assertResult(
+      Map(
+        1 -> Map(
+          AttributeName.withDefaultNS("int") -> AttributeNumber(2),
+          AttributeName.withDefaultNS("negInt") -> AttributeNumber(-2),
+          AttributeName.withDefaultNS("doubleRounded") -> AttributeNumber(2),
+          AttributeName.withDefaultNS("negDoubleRounded") -> AttributeNumber(-2),
+          AttributeName.withDefaultNS("double") -> AttributeNumber(2.34),
+          AttributeName.withDefaultNS("negDouble") -> AttributeNumber(-2.34)
+      ))) { unmarshalled }
+
+    val attrs = unmarshalled(1)
+
+    def assertScale(key: String, expectedScale: Int) = {
+      attrs(AttributeName.withDefaultNS(key)) match {
+        case num:AttributeNumber => assertResult(expectedScale, s"scale of $key attribute") {num.value.scale}
+        case _ => fail(s"expected AttributeNumber for $key")
+      }
+    }
+
+    assertScale("int", 0)
+    assertScale("negInt", 0)
+    assertScale("doubleRounded", 0)
+    assertScale("negDoubleRounded", 0)
+    assertScale("double", 2)
+    assertScale("negDouble", 2)
+  }
```

Comment:
I considered testing the json serialization, but it felt weird to include jsonformat stuff here, so I tested the lower-level detail (the scale of the BigDecimal).